eventapi.py

```diff
@@ -4,53 +4,53 @@
 from flask_restful import Resource, abort
 
 from event import Event
-from eventvalidations import *
+from eventvalidations import validate_put, validate_post
+from validations import validate_OId
 
 def read_args(request):
     return json.loads(request.data.decode('utf-8'))
 
 class EventAPI(Resource):
     def get(self, e_id):
+        if not validate_OId(e_id):
+            abort(404, message='Invalid id')
         event = Event.get_by_id(e_id)
+        if event is None:
+            abort(404, message='Event with id "{}" not found'.format(id_str))
         return vars(event)
 
     def put(self, e_id):
-        event = Event.get_by_id(e_id)
         data = read_args(request)
         try:
-            if 'from_time' in data:
-                read_time(data['from_time'])
-            if 'to_time' in data:
-                read_time(data['to_time'])
-            if 'days' in data:
-                validate_days(data['days'])
-            event.update(data)
-            validate_times(event.from_time, event.to_time)
-            return vars(event), 201
-
+            validate_put(e_id, data)
         except Exception as e:
             abort(404, message='Invalid payload to update event: {}'.format(e))
 
+        event = Event.get_by_id(e_id)
+        if event is None:
+            abort(404, message='Event with id "{}" not found'.format(id_str))
+
+        event.update(data)
+        return vars(event), 201
+
     def delete(self, e_id):
-        Event.delete(e_id)
+        if not validate_OId(e_id):
+            abort(404, message='Invalid id')
+        if not Event.delete(e_id):
```

Comment:
this should be within a try /except block .. the exception returned could also influence the return to be more than just `404`